components/script/dom/inputevent.rs

```diff
@@ -0,0 +1,85 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use dom::bindings::cell::DOMRefCell;
+use dom::bindings::codegen::Bindings::InputEventBinding::{self, InputEventMethods};
+use dom::bindings::codegen::Bindings::UIEventBinding::UIEventBinding::UIEventMethods;
+use dom::bindings::error::Fallible;
+use dom::bindings::js::{Root, RootedReference};
+use dom::bindings::reflector::reflect_dom_object;
+use dom::bindings::str::DOMString;
+use dom::uievent::UIEvent;
+use dom::window::Window;
+use dom_struct::dom_str;
+use std::cell::Cell;
+
+#[dom_struct]
+pub struct InputEvent {
+    uievent: UIEvent,
+    data: DOMRefCell<Option<DOMString>>,
+    is_composing: Cell<bool>,
```

Comment:
We can remove the DOMRefCell and Cell if we remove new_uninitialized and make new_inherited take the initial values.